src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol_Bases.cs

```diff
@@ -319,6 +319,12 @@ internal override ImmutableArray<NamedTypeSymbol> GetDeclaredInterfaces(ConsList
                 }
             }
 
+            if (declaration.Kind == DeclarationKind.Record)
+            {
+                var type = DeclaringCompilation.GetWellKnownType(WellKnownType.System_IEquatable_T);
+                baseInterfaces.Add(type.Construct(this));
```

Comment:
>baseInterfaces.Add [](start = 16, length = 18)

We should make sure we test error scenarios around this interface. Places like that usually try to locate matching syntax in the base list. Obviously, in this case there is no syntax to find. We should make sure compiler doesn't crash.